server/tests/groups.test.js

```diff
@@ -0,0 +1,93 @@
+import { itReturnsSuccess, run, query } from './common';
+
+describe('Get users group by passing filter', () => {
+  const response = query(`
+    {
+      user {
+        id
+        groups(id: 2) {
+          id
+          schedules {
+            id
+          }
+          events {
+            id
+          }
+          tags {
+            id
+          }
+        }
+      }
+    }
+  `);
+
+  itReturnsSuccess(response);
+  it('Returns correct single entity', () => response.then((res) => {
+    expect(res.data.user.groups.length).toBe(1);
+    expect(res.data.user.groups[0]).toHaveProperty('id');
+    expect(res.data.user.groups[0].id).toBe(2);
+    expect(res.data.user.groups[0].id).not.toBe(1);
+  }));
+});
+
+describe('Manipulate user group membership - leave', () => {
+  const response = run(
+    {
+      query: `
+      mutation removeUserFromGroup($groupUpdate: RemoveUserFromGroupInput!) {
+        removeUserFromGroup(groupUpdate: $groupUpdate)
+      }
+      `,
+      variables: {
+        groupUpdate: {
+          groupId: 2,
+        },
+      },
+    },
+  );
+
+  itReturnsSuccess(response);
+  it('should return true', () => response.then((res) => {
+    expect(res.data.removeUserFromGroup).toEqual(true);
+  }));
+});
+
+describe('Manipulate user group membership - join', () => {
```

Comment:
if these tests are failing sporadically you probably just need to put them into a single describe() so that they run together. Make sure you leave group after you've joined it, obviously.